### PR TITLE
Add Docker container for MSSQL for Linux

### PR DESCRIPTION
**Pull request by khalidabuhakmeh**
This let's folks run the benchmark regardless of their
host operating system. Previously it was SQLExpress.

🐳

### CODE REVIEW

**Comment by khalidabuhakmeh** [comment]
My benchmarks for funnzies. 

// * Summary *

BenchmarkDotNet=v0.12.1, OS=macOS 11.3.1 (20E241) [Darwin 20.4.0]
Intel Core i7-9750H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=6.0.100-preview.3.21202.5
  [Host]        : .NET Core 5.0.4 (CoreCLR 5.0.421.11614, CoreFX 5.0.421.11614), X64 RyuJIT
  .NET Core 5.0 : .NET Core 5.0.4 (CoreCLR 5.0.421.11614, CoreFX 5.0.421.11614), X64 RyuJIT

Job=.NET Core 5.0  Runtime=.NET Core 5.0  

|                 Method |     Mean |     Error |    StdDev |   Median | Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------- |---------:|----------:|----------:|---------:|------:|------:|------:|----------:|
|       DapperSystemData | 1.293 ms | 0.0258 ms | 0.0378 ms | 1.293 ms |     - |     - |     - |   3.63 KB |
|    DapperMicrosoftData | 1.343 ms | 0.0304 ms | 0.0822 ms | 1.330 ms |     - |     - |     - |   3.75 KB |
|    DapperAOTSystemData | 1.549 ms | 0.1045 ms | 0.2930 ms | 1.449 ms |     - |     - |     - |   2.06 KB |
| DapperAOTMicrosoftData | 1.373 ms | 0.0299 ms | 0.0839 ms | 1.366 ms |     - |     - |     - |   1.97 KB |

File: test/UsageBenchmark/Program.cs [hunk 1]
```diff
@@ -4,13 +4,33 @@
...
 using System;
 using System.Data.Common;
+using Ductus.FluentDocker.Builders;
 
 namespace UsageBenchmark
 {
     class Program
     {
+        
+        public static string ConnectionString
```
Comment (mgravell) [inline test/UsageBenchmark/Program.cs:14]: Nit: could be const?

File: test/UsageBenchmark/Program.cs [hunk 2]
```diff
@@ -4,13 +4,33 @@
...
+                    .UseImage("mcr.microsoft.com/mssql/server")
+                    .ExposePort(11433, 1433)
+                    .WithEnvironment("SA_PASSWORD=Pass123!", "ACCEPT_EULA=Y")
+                    .WaitForMessageInLog("Starting up database 'tempdb'.", TimeSpan.FromSeconds(30))
+                    .Build()
+                    .Start();
+
+            BenchmarkRunner.Run(typeof(Program).Assembly);
+            
+            container.Dispose();
```
Comment (mgravell) [inline test/UsageBenchmark/Program.cs:32]: This seems redundant - there's a `using` above

File: test/UsageBenchmark/Program.cs [hunk 3]
```diff
@@ -22,7 +42,8 @@ public class DataAccess : IDisposable
         [GlobalSetup]
         public void Connect()
         {
-            const string cs = @"Data Source=.\SQLEXPRESS;Initial Catalog=master;Integrated Security=True";
+            string cs = Program.ConnectionString;
```
Comment (mgravell) [inline test/UsageBenchmark/Program.cs:45]: Just use Program.ConnectionString in place of `cs`, i.e. nuke `cs`?

File: test/Dapper.AOT.Test/SamplesTests.cs [hunk 4]
```diff
@@ -23,15 +23,15 @@ where path.EndsWith(".input.cs", StringComparison.OrdinalIgnoreCase)
         [Theory, MemberData(nameof(GetFiles))]
         public void Run(string path)
         {
-            var intputPath = File.ReadAllText(path);
+            var inputPath = File.ReadAllText(path);
```
Comment (mgravell) [inline test/Dapper.AOT.Test/SamplesTests.cs:26]: 👍

File: test/Dapper.AOT.Test/SamplesTests.cs [hunk 5]
```diff
@@ -47,8 +47,8 @@ public void Run(string path)
             var actual = sb.ToString();
             try // automatically overwrite test output, for git tracking
             {
-                if (GetOriginCodeLocation() is string originFile
-                    && Path.GetDirectoryName(originFile) is string originFolder)
+                if (GetOriginCodeLocation() is { } originFile
+                    && Path.GetDirectoryName(originFile) is { } originFolder)
```
Comment (mgravell) [inline test/Dapper.AOT.Test/SamplesTests.cs:51]: nit: subjective, but I think I prefer the `is string` syntax (or `is not null` if we aren't capturing the value) when we're not using the `{...}` to test members - for clarity reasons

File: test/UsageBenchmark/Program.cs [hunk 6]
```diff
@@ -4,13 +4,33 @@
...
     {
+        
+        public static string ConnectionString
+            = "server=localhost,11433;database=master;user=sa;password=Pass123!;";
+        
         static void Main()
-            => BenchmarkRunner.Run(typeof(Program).Assembly);
+        {
+            using var container =
+                new Builder()
```
Comment (mgravell) [inline test/UsageBenchmark/Program.cs:20]: 👍 nice; will definitely use this approach for the integration test fixture, too

**Review by mgravell — CHANGES_REQUESTED** [review]
Hugely appreciated; that's an approach I haven't seen before, looks good! A couple of minor nits. The format changes - meh, I'm not precious, but I'll check whether we're going to run into reformat problems - but if so, I should probably add an editor config file (that's on me).

Can I also check that you're able and willing (and thus doing) to freely contribute code in accordance with the stated project license? (purely a formality)

**Comment by khalidabuhakmeh** [comment]
Ah! I was too slow. I'm going to close this PR and submit another PR. Seems some of the structure changed in the file and rebasing seem like a PITA.

**Pull request closed without merging**